pyxem/utils/segment_utils.py

```diff
@@ -0,0 +1,313 @@
+        except (RuntimeError, RuntimeWarning, ZeroDivisionError):
+            if np.all(f == np.zeros_like(f)) and np.all(t == np.zeros_like(t)):
+                corr = 1.
+            else:
+                corr = 0.
+
+    return corr
+
+
+def separate_watershed(vdf_temp, min_distance=1, min_size=1, max_size=np.inf,
+                       max_number_of_grains=np.inf, marker_radius=1,
+                       threshold=False, exclude_border=False, plot_on=False):
+    """Separate segments from one VDF image using edge-detection by the
+    sobel transform and the watershed segmentation implemented in
+    scikit-image. See [1,2] for examples from scikit-image.
+
+    Parameters
+    ----------
+    vdf_temp : np.array
+        One VDF image.
+    min_distance: int
+        Minimum distance (in pixels) between markers for them to be
+        considered separate markers for the watershed segmentation.
+    min_size : float
+        Grains with size (i.e. total number of pixels) below min_size
+        are discarded.
+    max_size : float
+        Grains with size (i.e. total number of pixels) above max_size
+        are discarded.
+    max_number_of_grains : int
+        Maximum number of grains included in the returned separated
+        grains. If it is exceeded, those with highest peak intensities
+        will be returned.
+    marker_radius : float
+        If 1 or larger, each marker for watershed is expanded to a disk
+        of radius marker_radius. marker_radius should not exceed
+        2*min_distance.
+    threshold : bool
+        If True, a mask is calculated by thresholding the VDF image by
+        the Li threshold method in scikit-image. If False (default), the
+        mask is the boolean VDF image.
+    exclude_border : int or True, optional
+        If non-zero integer, peaks within a distance of exclude_border
+        from the boarder will be discarded. If True, peaks at or closer
+        than min_distance of the boarder, will be discarded.
+    plot_on : bool
+        If True, the VDF, the mask, the distance transform
+        and the separated grains will be plotted in one figure window.
+
+    Returns
+    -------
+    sep : np.array
+        Array containing segments from VDF images (i.e. separated
+        grains). Shape: (image size x, image size y, number of grains)
+
+    References
+    ----------
+    [1] http://scikit-image.org/docs/dev/auto_examples/segmentation/
+        plot_watershed.html
+    [2] http://scikit-image.org/docs/dev/auto_examples/xx_applications/
+        plot_coins_segmentation.html#sphx-glr-auto-examples-xx-
+        applications-plot-coins-segmentation-py
+    """
+
+    # Create a mask from the input VDF image.
+    if threshold:
+        th = threshold_li(vdf_temp)
+        if np.isnan(th):
+            th = 0.
+        mask = np.zeros_like(vdf_temp)
+        mask[vdf_temp > th] = True
+    else:
+        mask = vdf_temp.astype('bool')
+    if np.any(np.nonzero(mask)) is False:
```

Comment:
What is the intention with this if statement? It looks a little odd to force a return None - wouldn't it be better to raise an error? In any case this condition needs a test.

Reply:
Will remove this checking. 

You've got to have put garbage in to be triggering this in, and as the saying goes, Garbage In Garbage Out.